pkg/overlay/cache.go

```diff
@@ -128,3 +131,18 @@ func (o *Cache) Put(ctx context.Context, nodeID storj.NodeID, value pb.Node) err
 
 	return o.DB.Put(nodeID.Bytes(), data)
 }
+
+// Delete will remove the node from the cache. Used when a node hard disconnects or fails
+// to pass a PING multiple times.
+func (o *Cache) Delete(ctx context.Context, id storj.NodeID) error {
```

Comment:
Why would we want to remove things from cache? I mean we can flag it as problematic or something instead.

Reply:
It's assumed that things in the cache are good to go. If we have a bad node in the cache it should be removed.

Reply:
Hard disconnect doesn't mean it's a bad node. E.g. losing network for half an hour would mean that the whole cache could be wiped.

Reply:
and then the cache would rebuild itself the next run. The assumption is everything in the cache was contacted successfully within the last hour. It's not a bad node if it's removed, it's a node we didn't successfully speak with.

Reply:
To add on to this: There are definitely cases where we want to remove a node from the cache, but also we aren't going to remove them after a single failure to communicate. I think we had talked in the past about removing nodes after X failed attempts, or X amount of time being unresponsive. This is just adding the method to the cache for that removal.